### ISSUE TITLE
Subsequent update failures on a successful request, removes the request id in the state file

### ISSUE THREAD

**Issue by markpeek** [terraform-providers/terraform-provider-vra7#20]
Issue transferred from old repo: https://github.com/vmware/terraform-provider-vra7/issues/96
Originally opened by: @Prativa20 

Failed update of a request should not remove the original request id from the terraform state file

**Comment by markpeek** [comment]
@skylerto would you happen to have a repro case for this issue?

**Comment by skylerto** [comment]
This one, I'm not sure -- I don't know how this translates into client requests.

**Comment by Prativa20** [comment]
I tried creating a scenario to fail update as follows:
1. In the BP, make the max mem as 10TB
2. Request the catalog item for the first time with 1 GB. VM is provisioned.
3. Update the mem in terraform config to 10TB and reprovision.
4. Update request is triggered but fails with error: Could not allocate the required storage or memory resources. Machine reservation vCenter Reservation does not have enough memory for reconfigure request

Observations:
1. The tf state file still retains the request id and the old data, which means the above issue is **not reproducible**. It must have been fixed when majority of the code was refactored.
2. However, the console was still showing that the operation is successful, which is a bug. I have fixed it in #22
